client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -670,7 +671,20 @@ class CancelPurchaseForm extends Component {
 					<BlankCanvas className="cancel-purchase-form">
 						<BlankCanvas.Header onBackClick={ this.closeDialog }>
 							{ this.getHeaderTitle() }
-							<span className="cancel-purchase-form__site-slug">{ site.slug }</span>
+							<span className="cancel-purchase-form__site-slug">
+								{ ( () => {
```

Comment:
I'd avoid the IIFEE here and instead move the switch to a instance method of the component class, i.e:

```js
    // feel free the tweak the method name
	getCancelSubjectDescription() {
		switch ( purchase.productId ) {
			// product.Id 5 is Domain Mapping
			case 5:
				return purchase.productName + translate( ' for ' ) + purchase.meta;
			// product.Id 36 is a Site Redirect
			case 36:
				return purchase.productName + translate( ' to ' ) + purchase.meta;
			default:
				return purchase.productName + translate( ' for ' ) + site.slug;
		}
	}
```

And then in the JSX:

```jsx
(...)
							<span className="cancel-purchase-form__site-slug">
								{ this.getCancelSubjectDescription() }
							</span
(...)

```

This would also improve code comprehension as it's easier to figure out what that piece of code is doing by reading the function name.

Reply:
That makes sense! This way it's something that is more modular as well, right? I'll implement this once we get some confirmation from Shilling about product IDs!